### PR TITLE
Fix findTags: Use query param to inject tags

### PR DESCRIPTION
**Pull request by marcelklehr**
Fixed the exception thrown in #366

### CODE REVIEW

**Comment by marcelklehr** [comment]
nc master is failing because of version spec in appinfo.

**Comment by blizzz** [comment]
>nc master is failing because of version spec in appinfo.
 
let me fix that

**Comment by blizzz** [comment]
can we have a test for this?

**Comment by blizzz** [comment]
Needs a rebase, tests are failing

**Comment by marcelklehr** [comment]
rebased.

**Comment by blizzz** [comment]
Still failing:

```
There was 1 failure:
1) OCA\Bookmarks\Tests\Test_LibBookmarks_Bookmarks::testFindTagsFilter
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
     0 => Array (
-        'tag' => 'four'
-        'nbr' => 1
+        'tag' => 'one'
+        'nbr' => '2'
     )
     1 => Array (
-        'tag' => 'one'
-        'nbr' => 2
+        'tag' => 'four'
+        'nbr' => '1'
     )
-    2 => Array (...)
 )

/home/travis/build/nextcloud/server/apps/bookmarks/tests/lib_bookmark_test.php:97
```

**Comment by blizzz** [comment]
I can still reproduce the same issue:

spectacle z13693

It also adds the tags "two", "three", "four" which are not used with the currently stored bookmarks. They somehow got stuck in the DB… perhaps not critical.

File: tests/lib_bookmark_test.php [hunk 1]
```diff
@@ -85,6 +85,17 @@ function testFindTags() {
...
 	}
+	
+	function testFindTagsFilter() {
+		$this->cleanDB();
+		$this->assertEquals($this->libBookmarks->findTags($this->userid), array());
+		$this->libBookmarks->addBookmark($this->userid, "http://www.google.de", "Google", array("one"), "PrivateNoTag", false);
+		$this->libBookmarks->addBookmark($this->userid, "http://www.heise.de", "Heise", array("one", "two"), "PrivatTag", false);
+		$this->libBookmarks->addBookmark($this->userid, "http://www.golem.de", "Golem", array("four"), "PublicNoTag", true);
+		
+		$findTags = $this->libBookmarks->findTags($this->userid, ["two"]);
```
Comment (blizzz) [inline tests/lib_bookmark_test.php]: you need to test with filtering two tags and assert that no tag is used multiple times and no wrong match is pulled it (cf. screenshot in my other comment).

**Comment by marcelklehr** [comment]
Ok. Testing with two tag filters, now. This doesn't fix the tags appear multiple times in the bookmark view, btw. It merely fixes the exception ;)

**Comment by blizzz** [comment]
It should fail then, the assertion is not strict enough

**Comment by marcelklehr** [comment]
I don't understand what you mean :?

**Comment by marcelklehr** [comment]
It shouldn't fail because tags appearing multiple times is not a problem of the find Tags method.

**Comment by blizzz** [comment]
D'oh!

**Pull request merged**